src/ngraph/runtime/cache.hpp

```diff
@@ -0,0 +1,63 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <list>
+#include <mutex>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include "ngraph/function.hpp"
+#include "ngraph/runtime/executable.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        class LRUCache : public std::enable_shared_from_this<LRUCache>
+        {
+        public:
+            using GraphCache = unordered_map<string, shared_ptr<Executable>>;
+            using ClonedFunctionMap = unordered_map<string, shared_ptr<Function>>;
+
+            LRUCache();
+
+            virtual ~LRUCache();
+
+            void add_entry(const vector<int>& shape,
```

Comment:
Why are shapes vector<int> instead of Shape?

Reply:
because shapes are of type size_t and my shapes have -1 in them as a separator between different args.
so if the input shapes are {2, 4} & {5,6,7} it would be stored as a key like {2, 4, -1, 5, 6, 7, -1}